### PR TITLE
Add basic skeleton for ssr capable search

### PR DESCRIPTION
**Pull request by PRTTMPRPHT**

### CODE REVIEW

File: next-app/src/app/search/page.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        console.error("Error fetching data:", error);
+    }
+
+    return (
+        <>
+            {
+                trails.map(trail => (<div key={trail._id.toString()}>{trail.name}</div>))
+            }
+        </>
+    );
```
Comment (PRTTMPRPHT) [inline next-app/src/app/search/page.tsx:37]: @malinjam This is a skeleton where you can add your UI. This component MUST NOT use the "use client" directive, so we also have to alter the Navbar to make it capable of being rendered server-side (first of all, not requiring these millions of props being passed around).

File: next-app/src/app/search/page.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import dbConnect from "@/lib/dbConnection";
+import {ServerTrailWithID, TrailModel} from "@/models/server/trail/trail";
+import {redirect} from "next/navigation";
+import "./page.scss";
+
+interface ServerSideProps {
+    searchParams?: {
+        l?: string; // Location.
+    }
+}
```
Comment (PRTTMPRPHT) [inline next-app/src/app/search/page.tsx:11]: effectively means:
`localhost:3000/search?l=Stockholm`

File: next-app/src/models/server/trail/trail.ts [hunk 3]
```diff
@@ -31,8 +32,16 @@ export class Trail implements Mutable<ServerTrail> {
...
+        this: ReturnModelType<typeof Trail>,
+        location: string,
+        withCreator: boolean = false
+    ): Promise<ServerTrailWithID[]> {
+        let query = this.find({location});
+        if (withCreator) {
+            query = query.populate("creator");
+        }
+        return await query.exec();
     }
```
Comment (PRTTMPRPHT) [inline next-app/src/models/server/trail/trail.ts:45]: @ferran98campos Let's maybe abolish `UserService` and put all the methods into `User` as static methods? It's kinda the recommended way by Typegoose, as the `UserModel` will basically inherit from `User`.

File: next-app/src/models/server/trail/trail.ts [hunk 4]
```diff
@@ -42,4 +51,5 @@ export class Trail implements Mutable<ServerTrail> {
...
 }
 
-export const TrailModel = getModelForClass(Trail);
+type TrailModelType = ReturnModelType<typeof Trail>;
+export const TrailModel: TrailModelType = mongoose.models.Trail as TrailModelType ?? getModelForClass(Trail);
```
Comment (PRTTMPRPHT) [inline next-app/src/models/server/trail/trail.ts:55]: This is important for having the hot reload mechanism during development.
The mongoose model will stay defined once it has been defined.
mongoose does not allow for re-definition.

This also means: Any modification of the model will require you to re-start your application.

**Review by ferran98campos — APPROVED** [review]

**Pull request merged**